aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1179,9 +1127,12 @@ public void setClientRegion(String clientRegion) {
    */
   public <T extends S3ClientBuilder> void applyS3CredentialConfigurations(T builder) {
```

Comment:
for these classes, my thought is that we can make them static methods taking multiple properties classes. This probably means we should have another PR to at least separate out the `HttpClientProperties` and `AwsClientProperties`. Before that, we can keep these methods as is for now.